src/main/java/Duke.java

```diff
@@ -1,10 +1,262 @@
+import duke.exception.AbsentArgsFlagException;
+import duke.exception.DukeException;
+import duke.exception.IllegalArgsNumException;
+import duke.exception.IllegalArgsTypeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.nio.Buffer;
+import java.util.Scanner;
+import java.util.Collections;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+
+
 public class Duke {
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
+        showLogo();
+        showWelcomeMsg();
+        readUserCmd();
+        showGoodbyeMsg();
+    }
+
+    private static void showLogo() {
+        String logo = " ____        _\n"
+                + "|  _ \\ _   _| | _____\n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+        System.out.println(logo);
+    }
+
+    private static final String END_CMD = "bye";
+    private static final String DEADLINE_FLAG = "/by";
+    private static final String EVENT_FLAG = "/at";
+    private static final String[] ILLEGAL_ARGS = {"blah"};
+    private static final int MAX_TASK = 100;
+    private static final int SEPARATOR_LEN = 50;
+    private static ArrayList<Task> tasks = new ArrayList<Task>();
+    private static final String DUKE_DUMP_FILE = "duke.txt";
+
+
+    private static void showSeparator() {
+        String separator = String.join("", Collections.nCopies(SEPARATOR_LEN, "="));
+        System.out.println(separator);
+    }
+
+    private static void showWelcomeMsg() {
+        showSeparator();
+        System.out.println("Hello! I'm Duke ^_^");
+        System.out.println("What can I do for you?");
+        showSeparator();
+    }
+
+    private static void showGoodbyeMsg() {
+        System.out.println("Bye. Hope to see you again soon! qwq");
+        showSeparator();
+    }
+
+    private static void showTaskCount() {
+        System.out.printf("Now we have %d tasks in the list.\n", tasks.size());
+        showSeparator();
+    }
+
+    private static void readUserCmd() {
+        Scanner sc = new Scanner(System.in);
+        String input = sc.nextLine();
+        while (!input.equals(END_CMD)) {
+            String[] args = input.split(" ", 2);
+            String arg1 = args[0];
```

Comment:
Here the use of arg1 and arg2 can be confusing. A better naming would be something like `String command` and `String userInput` to indicate the function of these two variables better. 

Code Quality Guideline: Name well